app/helpers/relationships_helper.rb

```diff
@@ -108,4 +108,14 @@ def relationship_form_tag(label, field, html_class = 'col-sm-10')
       label + content_tag(:div, class: html_class) { field }
     end
   end
+
+  # <Relationship> -> [ Struct ]
```

Comment:
This is cool to see structs popping up in the code! We can also use named structs like so (apologies if I'm telling you something you already know):

```ruby
# somewhere near top of file where we define things
Description = Struct.new(:title, :entity)

# in body here: 
titles_and_entities.map { |x| Description.new(*x) } 
```

I was thinking this might be useful elsewhere where we lean heavily on comments to define the fields of a hash but nowhere enforce them in code. Using structs might be a lightweight alternative to classes for doing that.

In this instance it seems like overkill since we don't reuse the struct... But in other instances where they're less fleeting and where the shape of the data was more integral to the logic of the computation (eg: all the `tag` lookup methods and network analysis methods where we do lots of hash manipulation) it might be worth it?

Reply:
yup yup...i didn't name the struct to save me a line of code :)

Reply:
yeah, totally, there's likely a bunch of cases where structs might suit us better than hashes; particularly to avoid the "is the key a string or a symbol or is it this a  hash with indifferent access thingy" question that happens all the time.

Reply:
> Is that not actually a bug?
> Does this commit address that bug in a way I'm not seeing?

I'm not sure i recall exactly...can you reproduce it? I remember thinking we thought there was a bug but then couldn't actually recreate the problem with kevin....